tsfc/impero_utils.py

```diff
@@ -88,14 +88,10 @@ def collect_temporaries(ops):
         # IndexSum temporaries should be added either at Initialise or
         # at Accumulate.  The difference is only in ordering
         # (numbering).  We chose Accumulate here.
-        if isinstance(op, (imp.Initialise, imp.Return, imp.ReturnAccumulate)):
-            pass
-        elif isinstance(op, imp.Accumulate):
+        if isinstance(op, imp.Accumulate):
             result.append(op.indexsum)
         elif isinstance(op, imp.Evaluate):
             result.append(op.expression)
-        else:
-            raise AssertionError("unhandled operation: %s" % type(op))
```

Comment:
I could have added `imp.Noop` to the first `if`, but then I decided to simplify the code by not listing all the accepted types, just the ones which are treated specially. This removes an assertion, but Impero nodes don't seem to change very often.